subprojects/execution/src/main/java/org/gradle/internal/execution/steps/CleanupOutputsStep.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.execution.steps;
+
+import com.google.common.collect.ImmutableSet;
+import org.gradle.internal.execution.IncrementalContext;
+import org.gradle.internal.execution.InputChangesContext;
+import org.gradle.internal.execution.Result;
+import org.gradle.internal.execution.Step;
+import org.gradle.internal.execution.UnitOfWork;
+import org.gradle.internal.execution.impl.OutputsCleaner;
+import org.gradle.internal.file.TreeType;
+import org.gradle.internal.fingerprint.FileCollectionFingerprint;
+import org.gradle.util.GFileUtils;
+
+import java.io.File;
+
+public class CleanupOutputsStep implements Step<InputChangesContext, Result> {
```

Comment:
For now I think this should be generic in both context and result to match the other steps. We can figure out how to clean up the situation later on.

Reply:
Most of the other steps seem to be only generic in the result or the context. The CacheStep does not have any generics. For me it doesn't seem we need to follow any consistency here.

Reply:
In any case it should be generic in its result as it doesn't create its own `Result` type.

Reply:
It is now generic in both types, given that it just passes throguh the context as well.